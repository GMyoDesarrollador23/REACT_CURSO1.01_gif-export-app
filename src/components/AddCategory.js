import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
   const [inputValue, setInputValue] = useState("");

   const handleInputValue = (e) => {
      setInputValue(e.target.value);
   };

   const handleSubmit = (e) => {
      e.preventDefault();
      // console.log("handleSubmit", inputValue);

      if (inputValue.trim().length > 2) {
         setCategories((cats) => [inputValue, ...cats]);

         setInputValue("");
      }
   };

   return (
      <form onSubmit={handleSubmit}>
         <p>{inputValue}</p>
         <input
            type="text"
            value={inputValue}
            onChange={handleInputValue}
         />
      </form>
   );
};

AddCategory.protoTypes = {
   setCategories: PropTypes.func.isRequired,
};
