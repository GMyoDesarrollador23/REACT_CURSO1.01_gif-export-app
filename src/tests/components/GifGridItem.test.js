import React from "react";
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";

describe("testeo de GifGridItem ", () => {
   const title = "titulo de prueba";
   const url = "https://algo.jpg";

   const wrapper = shallow(
      <GifGridItem title={title} url={url} />
   );

   // ---------------------------------------------------------
   test("debe de mostrar el componente correctamente", () => {
      expect(wrapper).toMatchSnapshot();
   });

   // ---------------------------------------------------------
   test("debe de tener un parrafo con el titulo", () => {
      const p = wrapper.find("p");
      expect(p.text().trim()).toBe(title);
   });

   // ---------------------------------------------------------
   test("debe de tener una imagen y url agul al pros", () => {
      const img = wrapper.find("img");
      // console.log(img.prop().src);
      expect(img.prop("src")).toBe(url);
      expect(img.prop("alt")).toBe(title);
   });

   // ---------------------------------------------------------
   test("debe tener la clase animate__fadeIn ", () => {
      const div = wrapper.find("div");
      const className = div.prop("className");

      expect(className.includes("animate__fadeIn")).toBe(
         true
      );
   });
   // ---------------------------------------------------------
});
