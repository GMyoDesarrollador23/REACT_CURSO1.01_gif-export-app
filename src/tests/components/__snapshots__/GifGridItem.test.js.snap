// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testeo de GifGridItem  debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo de prueba"
    src="https://algo,jpg"
  />
  <p>
    titulo de prueba
  </p>
</div>
`;
